### PR TITLE
update controller 

### PR DESCRIPTION
**Pull request by houz42**
- [x] use checkpointConditions in status
- [x] update checkpoint conditions only when job conditions changed

### CODE REVIEW

File: pkg/stub/handler.go [hunk 1]
```diff
@@ -16,149 +17,256 @@ import (
...
+	Verbose               bool   `json:"verbose,omitempty"`
 }
 
 func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 	switch event.Object.(type) {
 	case *v1alpha1.Checkpoint:
 		cp := event.Object.(*v1alpha1.Checkpoint)
 		if event.Deleted {
+			logger(cp).Info("deleting checkpoint")
+			if err := h.onDeletion(cp); err != nil {
```
Comment (killwing) [inline pkg/stub/handler.go]: 不用 delete

Comment (houz42) [reply pkg/stub/handler.go]: 删除绑定的 job, 就像删除 deployment 时候删除对应 replicaSet

Comment (killwing) [reply pkg/stub/handler.go]: 默认级联删除的

**Review by killwing — APPROVED** [review]
lgtm

**Pull request merged**